archive/display/src/main/scala/uk/ac/wellcome/platform/archive/display/DisplayProvider.scala

```diff
@@ -1,13 +1,62 @@
 package uk.ac.wellcome.platform.archive.display
 
-import io.circe.generic.extras.JsonKey
-import uk.ac.wellcome.platform.archive.common.progress.models.StorageProvider
+import io.circe.CursorOp.DownField
+import io.circe.{Decoder, DecodingFailure, Encoder, Json}
+import uk.ac.wellcome.platform.archive.common.progress.models.{
+  InfrequentAccessStorageProvider,
+  StandardStorageProvider,
+  StorageProvider
+}
+
+sealed trait DisplayProvider {
+  val id: String
+  final val ontologyType: String = "Provider"
+
+  def toStorageProvider: StorageProvider
+}
+
+case object StandardDisplayProvider extends DisplayProvider {
+  override val id: String = "aws-s3-standard"
 
-case class DisplayProvider(id: String,
-                           @JsonKey("type") ontologyType: String = "Provider") {
-  def toStorageProvider: StorageProvider = StorageProvider(id)
+  override def toStorageProvider: StorageProvider = StandardStorageProvider
 }
+
+case object InfrequentAccessDisplayProvider extends DisplayProvider {
+  override val id: String = "aws-s3-ia"
+
+  override def toStorageProvider: StorageProvider =
+    InfrequentAccessStorageProvider
+}
+
 object DisplayProvider {
   def apply(provider: StorageProvider): DisplayProvider =
-    DisplayProvider(id = provider.id)
+    provider match {
+      case StandardStorageProvider         => StandardDisplayProvider
+      case InfrequentAccessStorageProvider => InfrequentAccessDisplayProvider
+    }
+
+  implicit val decoder
+    : Decoder[DisplayProvider] = Decoder.instance[DisplayProvider](cursor =>
+    for {
+      id <- cursor.downField("id").as[String]
+      provider <- id match {
+        case StandardDisplayProvider.id => Right(StandardDisplayProvider)
+        case InfrequentAccessDisplayProvider.id =>
+          Right(InfrequentAccessDisplayProvider)
+        case _ =>
+          val fields = DownField("id") +: cursor.history
+          Left(DecodingFailure(
+            s"invalid value supplied, valid values are: ${StandardDisplayProvider.id}, ${InfrequentAccessDisplayProvider.id}.",
+            fields))
+      }
+    } yield {
+      provider
+  })
+
+  implicit val encoder: Encoder[DisplayProvider] =
+    Encoder.instance[DisplayProvider] { provider =>
+      Json.obj(
+        "id" -> Json.fromString(provider.id),
+        "type" -> Json.fromString("Provider"))
```

Comment:
Could you use the `ontologyType` on `DisplayProvider`? Otherwise there doesn't seem much of a reason to set it.

Reply:
Good point, I'll delete `ontologyType` cause I don't like it anyway